src/commands/containers/browseContainer.ts

```diff
@@ -0,0 +1,114 @@
+];
+
+function parsePortAndProtocol(portAndProtocol: string): { port: number, protocol: string | undefined } | undefined {
+    const splitPortAndProtocol = portAndProtocol.split('/');
+
+    if (splitPortAndProtocol.length > 0) {
+        const port = Number.parseInt(splitPortAndProtocol[0], 10);
+
+        if (!Number.isNaN(port)) {
+            return {
+                port,
+                protocol: splitPortAndProtocol.length > 1 ? splitPortAndProtocol[1].toLowerCase() : undefined
+            };
+        }
+    }
+
+    return undefined;
+}
+
+export async function browseContainer(context: IActionContext, node?: ContainerTreeItem): Promise<void> {
+    const telemetryProperties = <BrowseTelemetryProperties>context.telemetry.properties;
+
+    if (!node) {
+        node = await captureBrowseCancelStep('node', telemetryProperties, () => ext.containersTree.showTreeItemPicker<ContainerTreeItem>(ContainerTreeItem.allContextRegExp, context));
+    }
+
+    const inspectInfo = await node.getContainer().inspect();
+
+    const ports = inspectInfo.NetworkSettings && inspectInfo.NetworkSettings.Ports || {};
+    const possiblePorts =
+        Object.keys(ports)
+              .map(portAndProtocol => ({ mappings: ports[portAndProtocol], containerPort: parsePortAndProtocol(portAndProtocol) }))
+              // Exclude port designations we cannot recognized...
+              .filter(port => port.containerPort !== undefined)
+              // Exclude ports that are non-TCP-based (which should be the default)...
+              .filter(port => port.containerPort.protocol === undefined || port.containerPort.protocol === 'tcp')
+              // Exclude ports not mapped to the host...
+              .filter(port => port.mappings && port.mappings.length > 0);
+
+    telemetryProperties.possiblePorts = possiblePorts.map(port => port.containerPort.port);
+
+    if (possiblePorts.length === 0) {
+        // tslint:disable-next-line: no-floating-promises
+        ext.ui.showWarningMessage('No valid ports were mapped from the container to the host.');
+
+        return;
+    }
+
+    // Prefer common port (in order of preference)...
+    let selectedPort = possiblePorts.find(port => commonWebPorts.find(commonPort => commonPort === port.containerPort.port) !== undefined);
+
+    // Otherwise, if there's just a single port, assume that one...
+    if (selectedPort === undefined && possiblePorts.length === 1) {
+        selectedPort = possiblePorts[0];
+    }
+
+    // Otherwise, ask the user which port to use...
+    if (selectedPort === undefined) {
+        const items = possiblePorts.map(port => ({ label: port.containerPort.port.toString(), port }));
+
+        // Sort ports in ascending order...
+        items.sort((a, b) => a.port.containerPort.port - b.port.containerPort.port);
+
+        const item  = await captureBrowseCancelStep('port', telemetryProperties, () => ext.ui.showQuickPick(items, { placeHolder: 'Select the container port to browse to.' }));
```

Comment:
Nit: one extra space in `item  =`

Reply:
Surprised the linter didn't catch that, actually.